apps/opik-backend/src/main/java/com/comet/opik/domain/TraceDAO.java

```diff
@@ -695,4 +735,61 @@ public Mono<List<WorkspaceAndResourceId>> getTraceWorkspace(
                 .collectList();
     }
 
+    @Override
+    public Mono<Long> batchInsert(@NonNull List<Trace> traces, @NonNull Connection connection) {
+
+        Preconditions.checkArgument(!traces.isEmpty(), "traces must not be empty");
```

Comment:
Minor: non-null and empty can be done in one one by removing the non-null annotation and using `CollectionUtils.IsNotEmpty`.